### ISSUE TITLE
[BUG] Reliability issues with showing purchase dates

### ISSUE THREAD

**Issue by candela97** [tfedor/AugmentedSteam#998]
### Actual Behavior
Purchase dates don't always show, especially for games activated by keys.

From what I can gather, the problem partially lies with these replace regexes:
https://github.com/tfedor/AugmentedSteam/blob/3677e307ff22a3810003641aac4d4b6e003355b0/src/js/Background/Modules/SteamStoreApi.js#L126-L140
This list is largely unchanged from ES and needs an update. Examples of uncovered terms: "for Beta Testing", "- Steam Key Package", "Free Promotional Package - [Month Year]", "EU".

Other problems that prevent this feature from working properly:
a) Some package names have encoded "&" or spaces, some don't.
b) Sometimes store app names just don't match the package name due to localization or some other reason e.g. `https://store.steampowered.com/app/573170/`.

### Desired Behavior
For starters, the list of replace regexes needs to be updated. Or maybe there's a better way to get purchase dates?

**Comment by kazerniel** [comment]
I also have a few "complimentary reviewer package" ones that are keys devs send to Steam Curators. (They are not removable from the account, unlike Steam's own F2P/Complimentary licenses.)

**Comment by MxtOUT** [comment]
Yeah, I've also not been happy about how this is implemented.
But I suppose there is no other way to get the dates?

If we extend the list though, we might add some new false positives

**Comment by Revadike** [comment]
> But I suppose there is no other way to get the dates?

I got u bro ;)

https://help.steampowered.com/en/wizard/HelpWithGameIssue/?appid=440 or https://help.steampowered.com/en/wizard/HelpWithGameIssue/?appid=440&sessionid=SESSID&wizard_ajax=1

![image](https://user-images.githubusercontent.com/4411977/105845614-5b597d00-5fdb-11eb-832b-639468bfd5a7.png)

You can do this for every appid.

**Comment by Revadike** [comment]
Would be nice to parse refunds or revoke information too.

**Comment by candela97** [comment]
Yeah I tried Steam Support as well, but the date there sometimes doesn't match the one on your licenses page, especially if you used wallet funds or refunded the game then repurchased/activated again...

Definitely better than maintaining that list though.

**Comment by Revadike** [comment]
Uh, wdym it doesn't match? It just shows you everything. In fact, you can probably just copy-paste the whole text into the store page, if you request the locale page.

**Comment by MxtOUT** [comment]
That way of retrieving it is very nice!
And even if they don't match, it's data from the first party, it's their fault.

> Would be nice to parse refunds or revoke information too.

Yeah, I could imagine showing refund information and the source of your license (activated via key, part of sub/bundle)

**Comment by candela97** [comment]
> Uh, wdym it doesn't match? It just shows you everything. In fact, you can probably just copy-paste the whole text into the store page, if you request the locale page.

I meant stuff like this:
![螢幕擷取畫面 2021-01-28 212344](https://user-images.githubusercontent.com/54083835/106144598-4559d280-6129-11eb-9759-1ad0714011de.jpg)

If we only care about purchase dates, we can just copy the the date from ""Purchased", but I agree this feature can be expanded to show more info. Though for refunded items the review box may not appear on the store page.
